code/game/objects/items/weapons/tools/_tools.dm

```diff
@@ -898,6 +904,16 @@
 			ratio = max(round(ratio, 0.25) * 100, 25)
 			overlays += "[icon_state]-[ratio]"
 
+	if(ismob(loc))
+		var/tooloverlay
+		switch(mode)
+			if (EXCAVATE)
+				tooloverlay = "excavate"
+			if (DIG)
+				tooloverlay = "dig"
+		overlays += (tooloverlay)
```

Comment:
Why not move that into `pickaxe/update_icon()`?

Reply:
To easily allow this to be expanded over more tools if it is ever done in the future, also, I was unable to find a way to prevent `pickaxe/update_icon()` from cutting the overlays of `tools/update_icon()` without reiterating `tools/update_icon()` into `pickaxe/update_icon()`.